sqlglot/expressions.py

```diff
@@ -105,6 +105,45 @@ def to_s(self, level=0):
 
         return indent + left + right
 
+    def transform(self, fun, copy=True):
+        """
+        Recursively visits all tree nodes (excluding already transformed ones)
+        and applies the given transformation function to each node.
+
+        Args:
+            fun (function): a function which takes a node as an argument and returns a
+                new transformed node or the same node without modifications.
+            copy (bool): if set to True a new tree instance is constructed, otherwise the tree is
+                modified in place.
+
+        Returns:
+            the transformed tree.
+        """
+        node = deepcopy(self) if copy else self
+        new_node = fun(node)
+
+        if new_node is None:
+            raise ValueError('A transformed node cannot be None')
+        if not isinstance(new_node, Expression) or new_node is not node:
```

Comment:
new_node != node, what's the differenece with is?

Reply:
Later if we decide to introduce a proper equality comparison, I figured this way we won't have to compare entire subtrees. In other words I think this way is cheaper computation-wise. What do you think?

Reply:
ok sounds good